Mathlib/AlgebraicTopology/ModelCategory/Instances.lean

```diff
@@ -0,0 +1,359 @@
+/-
+Copyright (c) 2025 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.MorphismProperty.WeakFactorizationSystem
+import Mathlib.AlgebraicTopology.ModelCategory.CategoryWithCofibrations
+
+/-!
+# Consequences of model category axioms
+
+In this file, we deduce basic properties of fibrations, cofibrations,
+and weak equivalences from the axioms of model categories.
+
+-/
+
+
+universe w v u
+
+open CategoryTheory Limits MorphismProperty
+
+namespace HomotopicalAlgebra
+
+variable (C : Type u) [Category.{v} C]
+
+instance [CategoryWithWeakEquivalences C] [CategoryWithCofibrations C]
+    [(cofibrations C).IsStableUnderRetracts]
+    [(weakEquivalences C).IsStableUnderRetracts] :
+    (trivialCofibrations C).IsStableUnderRetracts := by
+  dsimp [trivialCofibrations]
+  infer_instance
+
+instance [CategoryWithWeakEquivalences C] [CategoryWithFibrations C]
+    [(fibrations C).IsStableUnderRetracts]
+    [(weakEquivalences C).IsStableUnderRetracts] :
+    (trivialFibrations C).IsStableUnderRetracts := by
+  dsimp [trivialFibrations]
+  infer_instance
+
+section IsStableUnderComposition
+
+variable {X Y Z : C} (f : X ⟶ Y) (g : Y ⟶ Z)
+
+instance [CategoryWithCofibrations C] [(cofibrations C).IsStableUnderComposition]
+    [hf : Cofibration f] [hg : Cofibration g] : Cofibration (f ≫ g) := by
+  rw [cofibration_iff] at hf hg ⊢
+  apply MorphismProperty.comp_mem <;> assumption
```

Comment:
```suggestion
    [hf : Cofibration f] [hg : Cofibration g] : Cofibration (f ≫ g) :=
  ⟨(cofibrations C).comp_mem _ _ hf.mem hg.mem⟩
```
We can shave a few lines in most properties by doing things like this.
I’d understand if you prefer going through `cofibration_iff` as a constructor, in this case
```suggestion
  cofibration_iff _|>.mpr <| (cofibrations C).comp_mem _ _ hf.mem hg.mem
```
is an option and uses less tactic, but this is arguably a matter of tastes here.

Reply:
I have made the change, but for such unexpensive proofs, I do not feel we should refrain from going in tactic mode.